tests/unit/server/api/v1/test_datasets.py

```diff
@@ -3183,19 +3183,45 @@ async def test_publish_dataset_without_fields(
         response = await async_client.put(f"/api/v1/datasets/{dataset.id}/publish", headers=owner_auth_header)
 
         assert response.status_code == 422
-        assert response.json() == {"detail": "Dataset cannot be published without fields"}
+        assert response.json() == {"detail": "Dataset cannot be published without required fields"}
+        assert (await db.execute(select(func.count(Record.id)))).scalar() == 0
+
+    async def test_publish_dataset_without_required_fields(
+        self, async_client: "AsyncClient", db: "AsyncSession", owner_auth_header: dict
+    ) -> None:
+        dataset = await DatasetFactory.create()
+        await TextFieldFactory.create(dataset=dataset)
+        await RatingQuestionFactory.create(dataset=dataset, required=True)
+
+        response = await async_client.put(f"/api/v1/datasets/{dataset.id}/publish", headers=owner_auth_header)
+
+        assert response.status_code == 422
+        assert response.json() == {"detail": "Dataset cannot be published without required fields"}
         assert (await db.execute(select(func.count(Record.id)))).scalar() == 0
 
     async def test_publish_dataset_without_questions(
         self, async_client: "AsyncClient", db: "AsyncSession", owner_auth_header: dict
     ):
         dataset = await DatasetFactory.create()
-        await TextFieldFactory.create(dataset=dataset)
+        await TextFieldFactory.create(dataset=dataset, required=True)
+
+        response = await async_client.put(f"/api/v1/datasets/{dataset.id}/publish", headers=owner_auth_header)
+
+        assert response.status_code == 422
+        assert response.json() == {"detail": "Dataset cannot be published without required questions"}
+        assert (await db.execute(select(func.count(Record.id)))).scalar() == 0
+
+    async def test_publish_dataset_without_required_questions(
+        self, async_client: "AsyncClient", db: "AsyncSession", owner_auth_header: dict
+    ) -> None:
+        dataset = await DatasetFactory.create()
+        await TextFieldFactory.create(dataset=dataset, required=True)
+        await RatingQuestionFactory.create(dataset=dataset)
 
         response = await async_client.put(f"/api/v1/datasets/{dataset.id}/publish", headers=owner_auth_header)
```

Comment:
We should just keep 1 of these unit tests, they are testing the same thing

Reply:
It was to ensure that the error is being raised on both no questions (former check) and no required questions (updated check)

Reply:
Maybe we can just parametrize the `required` value for both